devel/kokkos/Portfile

```diff
@@ -27,6 +27,15 @@ subport kokkos-devel {
                             size    2450033
     github.tarball_from     archive
     github.livecheck.branch develop
+    # Kokkos_HostSpace.cpp:79:11: error: aligned allocation function
+    # of type 'void *(std::size_t, std::align_val_t, const std::nothrow_t &)
+    # noexcept' is only available on macOS 10.13 or newer
+    if {${os.platform} eq "darwin" && ${os.major} < 17} {
```

Comment:
I think the check for clang should be here (or above), because gcc does not need this, it supports aligned allocation in `libstdc++`. The real issue is an outdated Apple `libc++`. (Yes, error message is misleading.)

Otherwise it looks like the port is broken on < 17 regardless, which is not the case.

P. S. Strictly speaking, the check should be for `libc++` and macOS version, and then for clang as a compiler, but perhaps it is too much: no one uses clangs with libstdc++ on macOS.

Reply:
P. S. And the title should include “with clang”.

Reply:
@barracuda156 this already looks good as is and doesn't affect systems with GCC by default. The simpler is better, isn't it?